lib/internal/modules/cjs/loader.js

```diff
@@ -700,6 +704,7 @@ Module._findPath = function(request, paths, isMain) {
 
   const selfFilename = trySelf(paths, exts, isMain, trailingSlash, request);
   if (selfFilename) {
+    emitExperimentalWarning('Resolve Self');
```

Comment:
For this one, perhaps _**Package name self resolution** is an experimental feature_. @jkrems does that work for you here?

Reply:
Sounds good! Thanks for all your work on this, @pd4d10!

Reply:
+1. Updated